stage1/init/init.go

```diff
@@ -122,8 +122,18 @@ func getArgsEnv(p *Pod, debug bool) ([]string, []string, error) {
 
 	case "src":
 		args = append(args, filepath.Join(common.Stage1RootfsPath(p.Root), nspawnBin))
-		args = append(args, "--boot") // Launch systemd in the pod
-
+		args = append(args, "--boot")              // Launch systemd in the pod
+		args = append(args, "--register", "false") // We cannot assume the host system is running systemd
+		out, err := os.Getwd()
+		if err != nil {
+			return nil, nil, err
+		}
+		lfd, err := common.GetRktLockFD()
+		if err != nil {
+			return nil, nil, err
+		}
+		args = append(args, fmt.Sprintf("--pid-file=%v", filepath.Join(out, "pid")))
+		args = append(args, fmt.Sprintf("--keep-fd=%v", lfd))
```

Comment:
In order for there to not be a hole between rkt putting the container into the "run" phase and it being visibly "running", rkt must acquire the lock prior to renaming the container into the run directory.  If it was left for nspawn to acquire the lock, the window between entering the "run" phase and the lock becoming acquired appears to be an exited container eligible for gc.